prov/shm/src/smr_ep.c

```diff
@@ -387,6 +408,94 @@ static int smr_progress_iov(struct smr_cmd *cmd, struct iovec *iov,
 	return -ret;
 }
 
+static void smr_do_atomic(void *src, void *dst, void *cmp, enum fi_datatype datatype,
+			  enum fi_op op, size_t cnt)
+{
+	char tmp_result[SMR_INJECT_SIZE];
+
+	if (op >= OFI_SWAP_OP_START) {
+		ofi_atomic_swap_handlers[op - OFI_SWAP_OP_START][datatype](dst,
+			src, cmp, tmp_result, cnt);
+	} else if (op != FI_ATOMIC_READ) {
+		ofi_atomic_write_handlers[op][datatype](dst, src, cnt);
+	}
+}
+
+static int smr_progress_inline_atomic(struct smr_cmd *cmd, struct fi_ioc *ioc,
+			       size_t ioc_count, size_t *len)
+{
+	int i;
+	uint8_t *src, *comp;
+
+
```

Comment:
I couldn't find anything else to complain about in this PR, so I'll complain that there's two blank lines here when only one is needed.

Reply:
This is so embarrassing. Thanks for catching that.